### ISSUE TITLE
Windows: conda-pack fails for packages with pip installed entry points

### ISSUE THREAD

**Issue by amangarg96** [conda/conda-pack#80]
I'm using conda-pack 0.3.1 on Windows 10 pro.

I created a conda environment which has a python package with entry points to scripts and conda-packed this environment.
When I untar it, run the conda-unpack.exe file and launch the script (to which the python package is pointing), it gives the following error:

`Fatal error in launcher: Unable to create process using '"c:\envs\testing-package-cp\python.exe"  "C:\Users\aman.garg\Downloads\testing-package\Scripts\bar.exe" ' 
`

Here, the path `c:\envs\testing-package-cp\python.exe` is pointing to python of the conda environment from which the tar was created 

This issue occurs only on Windows and not on Linux or MacOS.

**Comment by amangarg96** [comment]
To reproduce the issue, create the following python package:
- foo.py
```
import click

@click.group()
def bar():
    pass

@click.command()
def message():
    """Display the current version."""
    click.echo("Message!")

bar.add_command(message)
``` 

- setup.py
```
from setuptools import setup, find_packages
import utils 

setup(name='testing-package',
      version='0.0.2',
      entry_points={
        "console_scripts": ['bar = utils.foo:bar']
        }, 
      packages=find_packages(exclude=['contrib', 'docs', '*.tests'])
      )
```

Directory structure:
```
C:\USERS\AMAN.GARG\TESTING-PACKAGE
│   setup.py
└───utils
        foo.py
        __init__.py

```

**Comment by jcrist** [comment]
Hmmm, looks like the exe's provided by setuptools (e.g. things installed with pip) do need prefix rewriting, whereas those provided by conda packages don't (no absolute paths found in those). We currently don't do prefix rewriting for windows binaries, this would require some more work to handle. Is this a usecase that's important to you, and would you be willing/capable to submit a PR?

**Comment by amangarg96** [comment]
Yes, I would be willing to submit a PR. Could you guide me towards what needs to be done for handling the prefix for windows binaries?

**Comment by jcrist** [comment]
I'm not really sure what needs to be done, I wasn't aware that any absolute paths existed in the launcher exes on windows (and am not sure why the difference between packages). I'd need to read more/experiment a bit to give better advice. If you feel like looking into things, the relevant code is here: https://github.com/conda/conda-pack/blob/52daeddc54b2840b647e592032861c9b53fd3f28/conda_pack/prefixes.py#L100-L104

**Comment by amangarg96** [comment]
Thanks, I'll start probing into the code. 

Are the prefixes expected to be set during the conda-pack step? Or with the conda-unpack script?

**Comment by jcrist** [comment]
If a destination prefix is given, they should be replaced on packing, otherwise they should be recorded and fixed later in the `conda-unpack` script.

**Comment by Masseeh** [comment]
I tracked the issue and the problem is with case-sensitivity, therefore `replace_prefix` method should be changed to 
```python
def replace_prefix(data, mode, placeholder, new_prefix):
    if mode == 'text':
        data2 = text_replace(data, placeholder, new_prefix)
    elif mode == 'binary':
        data2 = binary_replace(data,
                               placeholder.encode('utf-8').lower(),
                               new_prefix.encode('utf-8').lower()
    return data2
```
I changed both of placeholder and new_prefix to lower case and also remove the line:
https://github.com/conda/conda-pack/blob/52daeddc54b2840b647e592032861c9b53fd3f28/conda_pack/prefixes.py#L84

**Comment by frascuchon** [comment]
> I tracked the issue and the problem is with case-sensitivity, therefore `replace_prefix` method should be changed to
> 
> ```python
> def replace_prefix(data, mode, placeholder, new_prefix):
>     if mode == 'text':
>         data2 = text_replace(data, placeholder, new_prefix)
>     elif mode == 'binary':
>         data2 = binary_replace(data,
>                                placeholder.encode('utf-8').lower(),
>                                new_prefix.encode('utf-8').lower()
>     return data2
> ```
> 
> I changed both of placeholder and new_prefix to lower case and also remove the line:
> https://github.com/conda/conda-pack/blob/52daeddc54b2840b647e592032861c9b53fd3f28/conda_pack/prefixes.py#L84

I confirm that it works !!! 

If there is no PR in process, I could create one with these changes

**Comment by achimnol** [comment]
I have encountered this issue and manually applying #100 has worked as expected.

**Comment by HiskiRobo** [comment]
If you need to fix this in a CI pipeline, i used the following to make it automated:
1) get patch.py here https://github.com/techtonik/python-patch/blob/master/patch.py
2) create separate build environment, install conda pack there, and run build from there (i called the env "builder", change from below if needed)

2) apply patch in your #100 as below before packing (in your batch file)
```
bitsadmin /transfer condapackpatch /download /priority normal https://github.com/conda/conda-pack/commit/b55c186e42134a33aa585fbd4723010805091997.patch %cd%\conda-pack.diff || goto :error
@CALL python .\patch.py conda-pack.diff --directory=.\envs\builder\lib\site-packages
```
3) run conda pack for needed environment(s) after this. The unpack command will then be patched for the packed envs.

**Comment by mcg1969** [comment]
Fixed by #122.
